### PR TITLE
Update js dependency and migrate to web

### LINKED ISSUE #670: Upgrade dependency js
There's a new version of the `js` package: https://github.com/dart-lang/sdk/blob/HEAD/pkg/js/CHANGELOG.md#070

Note that there is a breaking change:
> Breaking Change: Moved annotations to single location in dart:_js_annotations. This makes it easier to avoid collisions with dart:js_interop's @\JS, as you will now see a static error instead of it resolving in favor of the package definition. However, since this may result in breakages in previously working code, this is treated as a breaking change.

### LINKED ISSUE #679: Migrate to package:web
Flutter web apps are required to migrate from `dart:html` to `package:web` to use WASM.
This issue tracks the upgrade process for `flutter_secure_storage` .

See https://docs.flutter.dev/platform-integration/web/wasm and https://pub.dev/packages/web for more information.

### LINKED ISSUE #683: Update flutter_secure_storage_web depends on js ^0.6.3
Firstly, thank you for your work.

I'm having difficulty importing packages due to the broken js version.

I thank the attention.

`Because socket_io_client >=3.0.0-beta.0 depends on js ^0.7.1 and every version of flutter_secure_storage_web depends on js ^0.6.3, socket_io_client >=3.0.0-beta.0 is incompatible with flutter_secure_storage_web.`

### PR DESCRIPTION
**Pull request by ThexXTURBOXx**
This allows newer versions of `js` being used in conjunction with `flutter_secure_storage`. Also, as `html` is about to get phased out, this PR already migrates to `web` as it is recommended practice to do so.

Fixes #670 
Fixes #679 
Fixes #683

If you want to use this **now**, you can do so via a `dependency_override`:
```yaml
dependency_overrides:
  flutter_secure_storage:
    git:
      url: https://github.com/ThexXTURBOXx/flutter_secure_storage.git
      ref: develop
      path: flutter_secure_storage
```
You probably need to also override the other subpackages similarly (I won't explain how to do that. Just do it as I have shown with the main package above).

### CODE REVIEW

**Comment by ThexXTURBOXx** [comment]
This PR now also migrates from `html` to `web` since package maintainers are advised to do so sooner than later

**Comment by JgomesAT** [comment]
Is there any date to merge this and create a new version??

**Comment by ThexXTURBOXx** [comment]
@JgomesAT I don't even know if this will ever get merged as the repo owner is not very active currently.
For the time being, you can use my fork through a `dependency_override`

**Comment by ThexXTURBOXx** [comment]
You probably need to override the other packages as well

**Comment by JgomesAT** [comment]
> You probably need to override the other packages as well

What other package??

**Comment by ThexXTURBOXx** [comment]
As described above, the platform interface and web submodules (and probably also linux etc.

**Comment by JgomesAT** [comment]
> As described above, the platform interface and web submodules (and probably also linux etc.

but this library is part of the flutter_storage I don't have this library in the pubspec

**Comment by JgomesAT** [comment]
IF I put your repo in the pubspec dependencies: and the old one in the dependencies_override

dependencies:
   flutter_secure_storage:
       git:
         url: https://github.com/ThexXTURBOXx/flutter_secure_storage.git
         ref: develop
        path: flutter_secure_storage
        
        
dependency_overrides:  
  flutter_secure_storage: ^9.0.0
  
  I have  a successful build

**Comment by ThexXTURBOXx** [comment]
> IF I put your repo in the pubspec dependencies: and the old one in the dependencies_override

@JgomesAT Then, you are using the official version 

> but this library is part of the flutter_storage I don't have this library in the pubspec

You have all these packages in your `pubspec.lock` and hence you will need to override all of them.
Please refrain from commenting under this PR unless there is an issue with my code. For further information, please consult StackOverflow instead.

**Comment by JgomesAT** [comment]
I think that is an issue of your code I only use your repo, and I removed all  references of package flutter_secure_storage: ^9.0.0
the error is in the run in a flutter in a web in debug mode in an Apple(iMAC) device

**Comment by ThexXTURBOXx** [comment]
Exactly, after merging and releasing these changes properly to pub.dev, this won't be necessary anymore.

File: flutter_secure_storage_web/lib/src/subtle.dart
Comment (IchordeDionysos) [inline flutter_secure_storage_web/lib/src/subtle.dart:25]: The migration to the new js-interop approach also means getting rid of the `package:js`.

https://dart.dev/interop/js-interop/package-web

See also these comments https://github.com/firebase/flutterfire/issues/12027#issue-2038508266 from Kevin Moore (Flutter Web PM)

File: flutter_secure_storage_web/pubspec.yaml
Comment (IchordeDionysos) [inline flutter_secure_storage_web/pubspec.yaml:16]:
```suggestion
```

The migration to the new js-interop approach also means getting rid of the package:js.

https://dart.dev/interop/js-interop/package-web

See also these comments https://github.com/firebase/flutterfire/issues/12027#issue-2038508266 from Kevin Moore (Flutter Web PM)

**Review by IchordeDionysos — CHANGES_REQUESTED** [review]

File: flutter_secure_storage_web/pubspec.yaml
Comment (ThexXTURBOXx) [reply flutter_secure_storage_web/pubspec.yaml:16]: Oh yes, thanks for reminding me about this... Currently, I don't have enough free time to also address this, but hopefully I will do so soon!

Comment (IchordeDionysos) [reply flutter_secure_storage_web/pubspec.yaml:16]: No worries 😌 I hear your struggle maintaining open-source projects!

**Comment by ThexXTURBOXx** [comment]
I think yes. Someone might want to migrate some stuff to `js_interop`, though

**Comment by jtavio** [comment]
When will you make the merge of this?

**Comment by ThexXTURBOXx** [comment]
@juliansteenbakker Since I have seen that you are about to incorporate this, I want to mention that the js_interop migration from #698 should also be considered. A mixture of this PR right here and #698 is probably the best way to go.
Do you want me to rebase my PR right here onto your upstream master?

**Comment by juliansteenbakker** [comment]
Hi @ThexXTURBOXx , that would be awesome ! I am planning on releasing  a smaller version with bug fixes first today, and a new major version with some breaking changes and possibly this PR somewhere in the end of this week.

**Comment by ThexXTURBOXx** [comment]
@juliansteenbakker Done! :)

File: flutter_secure_storage_web/pubspec.yaml
Comment (juliansteenbakker) [inline flutter_secure_storage_web/pubspec.yaml:7]: Everything looks good, however the only thing that's a bit annoying is that the web package is available from dart 3.3, which is basically the latest flutter version, meaning that if we publish this, everyone is required to run the latest Flutter version.

Comment (ThexXTURBOXx) [reply flutter_secure_storage_web/pubspec.yaml:7]: Yes, this is true. I have pushed the constraints for `web` down as far as possible. Older versions lack features that are (IMHO) critical for the functionality of `flutter_secure_storage`. Maybe, the constraints *could* be pushed even lower than that using some hacky workarounds, but I am unsure.
This is also one of the reasons why it makes sense to apply this PR (and #698 for that matter) only after a major version bump

**Comment by juliansteenbakker** [comment]
I have released v9.1.0 which has all fixes that are possible for the current lower sdk version. So i will now merge this, and the next release (v10) will use the higher sdk version.

**Pull request merged**
